src/Kestrel.Core/Internal/Http/HttpResponseStream.cs

```diff
@@ -145,25 +146,23 @@ public void Abort()
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void ValidateState(CancellationToken cancellationToken)
         {
-            switch (_state)
+            var state = _state;
+            if (state == HttpStreamState.Open || state == HttpStreamState.Aborted)
             {
-                case HttpStreamState.Open:
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        cancellationToken.ThrowIfCancellationRequested();
-                    }
-                    break;
-                case HttpStreamState.Closed:
-                    throw new ObjectDisposedException(nameof(HttpResponseStream), CoreStrings.WritingToResponseBodyAfterResponseCompleted);
-                case HttpStreamState.Aborted:
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        // Aborted state only throws on write if cancellationToken requests it
-                        cancellationToken.ThrowIfCancellationRequested();
-                    }
-                    break;
+				// Aborted state only throws on write if cancellationToken requests it
+				cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Do we know if CancellationToken.ThrowIfCancellationRequested itself gets inlined?

Reply:
Yes it gets inlined. Checked the dasm. And it's just the same property-access, so this duplication was removed. See https://github.com/dotnet/coreclr/blob/85374ceaed177f71472cc4c23c69daf7402e5048/src/System.Private.CoreLib/src/System/Threading/CancellationToken.cs#L305-L311